app/src/main/java/org/simple/clinic/drugs/search/DrugModule.kt

```diff
@@ -11,4 +21,15 @@ object DrugModule {
   fun drugDao(database: AppDatabase): Drug.RoomDao {
     return database.drugDao()
   }
+
+  @Provides
+  fun drugSyncApi(@Named("for_country") retrofit: Retrofit): DrugSyncApi {
+    return retrofit.create(DrugSyncApi::class.java)
+  }
+
+  @Provides
+  @TypedPreference(LastDrugPullToken)
```

Comment:
I've been considering creating resource scoped modules for DI. An example is the `@ForPatientSync` qualifier, which is currently being used only for getting the record retention fallback duration, but I am planning to move all the DI qualifiers related to patient sync to this and refactoring the DI setup around resource scope,

What do you think of this approach?

Reply:
Yeah, that would be really useful and easy to understand

Reply:
https://app.clubhouse.io/simpledotorg/story/4126/refactor-sync-related-di-providers-into-resource-scoped-modules